addon/webextension/selector/ui.js

```diff
@@ -322,11 +322,98 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
 
   iframePreSelection.onResize = watchFunction(onResize.bind(iframePreSelection));
 
+  let iframePreview = exports.iframePreview = {
+    element: null,
+    document: null,
+    sizeTracking: {
+      windowDelayer: null
+    },
+    display(installHandlerOnDocument, standardOverlayCallbacks) {
+      return new Promise((resolve, reject) => {
+        if (!this.element) {
+          this.element = document.createElement("iframe");
+          this.element.src = browser.extension.getURL("blank.html");
+          this.element.id = "firefox-screenshots-preview-iframe";
+          this.element.style.zIndex = "99999999999";
+          this.element.style.border = "none";
+          this.element.style.position = "fixed";
+          this.element.style.top = "0";
+          this.element.style.left = "0";
+          this.element.style.margin = "0";
+          this.element.scrolling = "no";
+          this.updateElementSize();
+          this.element.onload = watchFunction(() => {
+            this.document = this.element.contentDocument;
+            this.document.documentElement.innerHTML = `
+              <head>
+                <style>${substitutedCss}</style>
+                <title></title>
+              </head>
+              <body>
+                <div class="preview-overlay">
+                  <div class = "preview-image">
+                    <div class = "preview-buttons">
+                      <button class = "highlight-button-cancel"></button>
+                      <button class = "highlight-button-download"></button>
+                      <button class = "highlight-button-save"></button>
+                    </div>
+                  </div>
+                </div>
+              </body>`;
+            installHandlerOnDocument(this.document);
+            const overlay = this.document.querySelector(".preview-overlay");
+            overlay.querySelector(".highlight-button-save").textContent = browser.i18n.getMessage("saveScreenshotSelectedArea");
```

Comment:
It might be good to use the code/approach from slides.js – the code that does the i18n substitution could get pulled out and used here.